### PR TITLE
fix title, translate with slot

### LINKED ISSUE #831: Inconsistent renaming "Contract Definition" to "Data Offer"
### Description - What happened? *

```
- Issue 1 : EDC UI dashboard page we have name mismatch Data offer and contract definitions
- Issue 2 : Incorrect error message displayed "No contract definition found with given filter" when search data offer is not available
- Issue 3 : When we publish new data offer(contract definitions) incorrect error message is displayed
```

### Expected Behavior *

Renaming "Contract Definition" to "Data Offer" in all places

### Observed Behavior *

Inconsistent renaming "Contract Definition" to "Data Offer", see Screenshots for exmaples.

### Steps to Reproduce

_No response_

### Context Information

_No response_

### Relevant log output

_No response_

### Screenshots

image

image

image

### PR DESCRIPTION
**Pull request by illfixit**
_What issues does this PR close?_
fix title, translate with slot,

Titlebar is working
image

Text on the dashboard is working
image

```[tasklist]
### Checklist
- [x] The PR title is short and expressive.
- [ ] I have updated the CHANGELOG.md. See [changelog_update.md](https://github.com/sovity/authority-portal/tree/main/docs/dev/changelog_updates.md) for more information.
- [ ] I have updated the Deployment Migration Notes Section in the CHANGELOG.md for any configuration / external API changes.
- [x] I have performed a **self-review**
```

### CODE REVIEW

File: src/app/routes/connector-ui/connector-ui.component.ts [hunk 1]
```diff
@@ -34,7 +34,12 @@ export class ConnectorUiComponent implements OnInit {
   ) {
     this.navItemGroups = this.navItemsBuilder.buildNavItemGroups();
     this.translateService.setDefaultLang('en');
-    this.translateService.use(localStorage.getItem('selectedLanguage') || 'en');
+    let selectedLanguage = localStorage.getItem('selectedLanguage');
+    if (!selectedLanguage) {
+      selectedLanguage = 'en';
+      localStorage.setItem('selectedLanguage', selectedLanguage);
+    }
+    this.translateService.use(JSON.parse(selectedLanguage));
```
Comment (richardtreier) [inline src/app/routes/connector-ui/connector-ui.component.ts:42]: prefer LocalStoredValue code from LanguageSelector, move that code rather to a service than do custom LocalStorage interactions

File: src/app/shared/common/translate-with-slot/translate-with-slot.component.ts [hunk 2]
```diff
@@ -22,6 +24,14 @@ export class TranslateWithSlotComponent implements OnChanges, OnDestroy {
   textAfter = '';
 
   constructor(private translationService: TranslateService) {
+    this.splitText();
+  }
+
+  ngOnInit() {
+    this.splitText();
+  }
```
Comment (richardtreier) [inline src/app/shared/common/translate-with-slot/translate-with-slot.component.ts:32]: keep one of the two

**Review by richardtreier — CHANGES_REQUESTED** [review]

File: src/app/shared/common/translate-with-slot/translate-with-slot.component.ts (hunk 2 above)
Comment (illfixit) [reply src/app/shared/common/translate-with-slot/translate-with-slot.component.ts:32]: it didn't work with only one

Comment (richardtreier) [reply src/app/shared/common/translate-with-slot/translate-with-slot.component.ts:32]: maybe replace translateService.get with translateService.stream

Comment (illfixit) [reply src/app/shared/common/translate-with-slot/translate-with-slot.component.ts:32]: looks like stream with only constructor works great, thanks!

**Review by richardtreier — CHANGES_REQUESTED** [review]

**Review by richardtreier — APPROVED** [review]
LGTM

**Pull request merged**